sync-core/src/main/java/com/cloudant/sync/datastore/Datastore.java

```diff
@@ -267,7 +268,7 @@ public void resolveConflictsForDocument(String docId, ConflictResolver resolver)
      * @throws IOException if there is a problem saving any new attachments
      * @see Datastore#getEventBus()
      */
-    public BasicDocumentRevision createDocumentFromRevision(MutableDocumentRevision rev) throws IOException;
+    public BasicDocumentRevision createDocumentFromRevision(MutableDocumentRevision rev) throws IOException, DocumentException;
```

Comment:
Or to state another way, should updateDocumentFromRevision just throw DocumentException?